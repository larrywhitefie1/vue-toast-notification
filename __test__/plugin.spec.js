import {mount, createLocalVue} from '@vue/test-utils'
import Plugin from '../src/index';

describe('Toast plugin', () => {

  // Make a copy of local vue
  let localVue = createLocalVue();
  // Define the global component
  localVue.use(Plugin);

  test('has instance registered', () => {
    expect(localVue.$toast).toBeDefined()
  });

  test('has open method on instance', () => {
    expect(localVue.$toast.open).toBeDefined()
  });

  test('has dismiss method on instance returned by open()',async () => {
    let toast = localVue.$toast.open('<b>Sample</b>');
    await toast.$nextTick();

    expect(toast.$el).toMatchSnapshot();
    expect(toast.dismiss).toBeDefined();
    toast.dismiss();
  });

});
