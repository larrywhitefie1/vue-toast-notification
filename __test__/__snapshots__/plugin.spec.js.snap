// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast plugin has dismiss method on instance returned by open() 1`] = `
<div
  class="v-toast v-toast-success is-bottom-right v-enter fadeInUp"
  role="alert"
  style=""
>
  <div
    class="v-toast-icon"
  />
   
  <p
    class="v-toast-text"
  >
    <b>
      Sample
    </b>
  </p>
</div>
`;
